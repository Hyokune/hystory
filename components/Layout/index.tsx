import Head from 'next/head';
import React, { Fragment } from 'react';
import Header from '../Header';

interface Props {
  children: JSX.Element[] | JSX.Element,
}

const Layout = ({ children }: Props) => (
  <Fragment>
    <Head>
      <title>Hystory</title>
      <link rel="stylesheet" type="text/css" href="https://cdnjs.cloudflare.com/ajax/libs/meyer-reset/2.0/reset.min.css" />
      <link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.13.0/css/all.css" integrity="sha384-Bfad6CLCknfcloXFOyFnlgtENryhrpZCe29RTifKEixXQZ38WheV+i/6YWSzkz3V" crossOrigin="anonymous" />
    </Head>
    <Header />
    <main>
      {children}
    </main>
  </Fragment>
);

export default Layout;
