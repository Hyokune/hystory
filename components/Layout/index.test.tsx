import { shallow } from 'enzyme'
import React from 'react'
import Layout from './index'

const childElement = <p>Child element</p>

describe('Layout component', () => {
  test('render component without error', () => {
    const wrapper = shallow(<Layout>{childElement}</Layout>)
    expect(wrapper).toMatchSnapshot()
  })
})
