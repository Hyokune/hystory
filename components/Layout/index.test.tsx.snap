// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout component render component without error 1`] = `
<Fragment>
  <Head>
    <title>
      Hystory
    </title>
    <link
      href="https://cdnjs.cloudflare.com/ajax/libs/meyer-reset/2.0/reset.min.css"
      rel="stylesheet"
      type="text/css"
    />
    <link
      crossOrigin="anonymous"
      href="https://use.fontawesome.com/releases/v5.13.0/css/all.css"
      integrity="sha384-Bfad6CLCknfcloXFOyFnlgtENryhrpZCe29RTifKEixXQZ38WheV+i/6YWSzkz3V"
      rel="stylesheet"
    />
  </Head>
  <Header />
  <main>
    <p>
      Child element
    </p>
  </main>
</Fragment>
`;
